vita3k/modules/SceHttp/SceHttp.cpp

```diff
@@ -303,35 +1045,153 @@ EXPORT(int, sceHttpSslIsCtxCreated) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceHttpUriBuild) {
+EXPORT(int, sceHttpUriBuild, char *out, unsigned int *require, unsigned int prepare, const SceHttpUriElement *srcElement, unsigned int option) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (!out || !srcElement)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
+    // TODO
+
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceHttpUriEscape) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceHttpUriEscape, char *out, unsigned int *require, unsigned int prepare, const char *in) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (!out || !in)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
+    // TODO: test
+
+    auto encoded = net_utils::encodeURIComponent(std::string(in)).c_str();
+
+    strcpy(out, encoded);
+
+    return 0;
 }
 
-EXPORT(int, sceHttpUriMerge) {
+EXPORT(int, sceHttpUriMerge, char *mergedUrl, const char *url, const char *relativeUrl, unsigned int *require, unsigned int prepare, unsigned int option) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (!mergedUrl || !relativeUrl)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
+    if (!url)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_URL);
+
+    // TODO: need example from game
+
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceHttpUriParse) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceHttpUriParse, SceHttpUriElement *out, const char *srcUrl, void *pool, size_t *require, size_t prepare) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (!srcUrl)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_URL);
+
+    if (!out)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
+    auto parsed = net_utils::parse_url(srcUrl);
+
+    if (parsed.invalid)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_URL);
+
+    out->opaque = false;
+    out->scheme = parsed.scheme;
+    out->username = parsed.username;
+    out->password = parsed.password;
+    out->hostname = parsed.hostname;
+    out->path = parsed.path;
+    out->query = parsed.query;
+    out->fragment = parsed.fragment;
+    auto port = std::stoi(parsed.port);
+    out->port = (unsigned short)port;
+
+    return 0;
 }
 
-EXPORT(int, sceHttpUriSweepPath) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceHttpUriSweepPath, char *dst, const char *src, size_t srcSize) {
+    if (!host.http.inited)
+        return RET_ERROR(SCE_HTTP_ERROR_BEFORE_INIT);
+
+    if (!dst)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_VALUE);
+
+    if (!src)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_URL);
+
+    auto parsed = net_utils::parse_url(src);
+
+    if (parsed.invalid)
+        return RET_ERROR(SCE_HTTP_ERROR_INVALID_URL);
+
+    char *credentials = { 0 };
+    if (parsed.username) { // we have credentials
+        strcat(credentials, parsed.username);
```

Comment:
stack overflow, you need to allow enough memory to credentials to contains the whole string (right now only one byte is allocated)

Reply:
Your fix half-solved the issue, credentials is not initialized and contain random data, zero-fill it first (or at least the first char should be enough).